server/src/storage/segment.rs

```diff
@@ -58,15 +58,15 @@ pub struct SegmentBuilder {
 impl SegmentBuilder {
     pub fn new() -> Self {
         Self {
-            timestamps: TimestampBuilder::new(INITIAL_BUILDER_CAPACITY),
-            values: ValueBuilder::new(INITIAL_BUILDER_CAPACITY),
+            timestamps: TimestampBuilder::new(BUILDER_CAPACITY),
+            values: ValueBuilder::new(BUILDER_CAPACITY),
         }
     }
 
     /// Return the total size of the builder in bytes. Note that this is constant.
     pub fn get_memory_size() -> usize {
-        (INITIAL_BUILDER_CAPACITY * mem::size_of::<Timestamp>())
-            + (INITIAL_BUILDER_CAPACITY * mem::size_of::<Value>())
+        (BUILDER_CAPACITY * mem::size_of::<Timestamp>())
```

Comment:
Since this is a constant when compiled it does not seem like there would be any performance benefits to this. In that case I would prefer keeping it like this to avoid cluttering the types file.